src/Features/Core/Portable/Snippets/SnippetProviders/AbstractSnippetProvider.cs

```diff
@@ -205,5 +213,10 @@ private async Task<Document> AddFormatAnnotationAsync(Document document, int pos
             document = document.WithSyntaxRoot(annotatedSnippetRoot);
             return document;
         }
+
+        protected virtual async Task<Document> AddIndentationToDocumentAsync(Document document, int position, ISyntaxFacts syntaxFacts, CancellationToken cancellationToken)
```

Comment:
Add some xml docs on why this exists, when it's called, and how this relates to formatting